In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('tuesday_3.csv')
df.head()

Unnamed: 0  Alcohol                                   Appellation Category  \
0           0     14.9  Russian River Valley, Sonoma, California, US      Red   
1           1     14.5  Central Coast, Central Coast, California, US      Red   
2           2     13.8                           California Republic      Red   
3           3     13.8          Lodi, Central Valley, California, US      Red   
4           4     13.0                                New Jersey, US      Red   

   Points  Price                                             Review  \
0      87  42.00  Baked plum, licorice and lavender aromas and f...   
1      87  19.98  Pomegranate and light baking-spice aromas show...   
2      86  15.00  Butter and vanilla notes dominate the jammy fr...   
3      86  10.00  An aroma like toasted almonds and wood smoke m...   
4      86  27.00  Plum skin and pomegranate lead the nose while ...   

                                                 Url     Variety  \
0  https://www.winemag.com/buying-guide/v-sattui-...   Zinfandel   
1  https://www.winemag.com/buying-guide/wente-201...  Pinot Noir   
2  https://www.winemag.com/buying-guide/californi...  California   
3  https://www.winemag.com/buying-guide/collier-c...  Pinot Noir   
4  https://www.winemag.com/buying-guide/diluca-20...   Red Blend   

                                                Wine         ...          \
0  V. Sattui 2015 Gilsson Vineyard Old Vine Zinfa...         ...           
1  Wente 2014 Coastal Selection Pinot Noir (Centr...         ...           
2  California Republic 2016 Cabernet Sauvignon (C...         ...           
3     Collier Creek 2015 Red Wagon Pinot Noir (Lodi)         ...           
4                  DiLuca 2016 Rosso Black Label Red         ...           

           Appellation1     Appellation2 Appellation3 Appellation4  \
0  Russian River Valley           Sonoma   California           US   
1         Central Coast    Central Coast   California           US   
2   California Republic              NaN          NaN          NaN   
3                  Lodi   Central Valley   California           US   
4            New Jersey               US          NaN          NaN   

                             wine_type   Varietals      Region    Year  \
0  Gilsson Vineyard Old Vine Zinfandel   Zinfandel         NaN  2015.0   
1         Coastal Selection Pinot Noir  Pinot Noir         NaN  2014.0   
2                   Cabernet Sauvignon         NaN  California  2016.0   
3                 Red Wagon Pinot Noir  Pinot Noir         NaN  2015.0   
4                    Rosso Black Label   Red Blend         NaN  2016.0   

                Region2                 Var  
0  Russian River Valley           Zinfandel  
1         Central Coast          Pinot Noir  
2            California  Cabernet Sauvignon  
3                  Lodi          Pinot Noir  
4                   NaN           Red Blend  

[5 rows x 21 columns]

In [3]:
df.describe(include= 'all')


Unnamed: 0      Alcohol  \
count   4768.000000  4768.000000   
unique          NaN          NaN   
top             NaN          NaN   
freq            NaN          NaN   
mean    2485.517827    13.949778   
std     1439.985066     0.950407   
min        0.000000     8.000000   
25%     1227.750000    13.500000   
50%     2493.500000    14.100000   
75%     3732.250000    14.500000   
max     4973.000000    22.000000   

                                         Appellation Category       Points  \
count                                           4768     4768  4768.000000   
unique                                           945        7          NaN   
top     Russian River Valley, Sonoma, California, US      Red          NaN   
freq                                             214     3118          NaN   
mean                                             NaN      NaN    89.340185   
std                                              NaN      NaN     2.784258   
min                                              NaN      NaN    80.000000   
25%                                              NaN      NaN    87.000000   
50%                                              NaN      NaN    90.000000   
75%                                              NaN      NaN    91.000000   
max                                              NaN      NaN   100.000000   

              Price                                             Review  \
count   4768.000000                                               4768   
unique          NaN                                               4678   
top             NaN  This is an attractive well-made dry wine that ...   
freq            NaN                                                  2   
mean      42.290271                                                NaN   
std       32.311230                                                NaN   
min        5.000000                                                NaN   
25%       24.000000                                                NaN   
50%       35.000000                                                NaN   
75%       50.000000                                                NaN   
max      475.000000                                                NaN   

                                                      Url     Variety  \
count                                                4768        4768   
unique                                               4678         228   
top     https://www.winemag.com/buying-guide/jason-ste...  Pinot Noir   
freq                                                    2         862   
mean                                                  NaN         NaN   
std                                                   NaN         NaN   
min                                                   NaN         NaN   
25%                                                   NaN         NaN   
50%                                                   NaN         NaN   
75%                                                   NaN         NaN   
max                                                   NaN         NaN   

                                               Wine     ...      \
count                                          4768     ...       
unique                                         4676     ...       
top     Naggiar 2017 Estate Rosé (Sierra Foothills)     ...       
freq                                              3     ...       
mean                                            NaN     ...       
std                                             NaN     ...       
min                                             NaN     ...       
25%                                             NaN     ...       
50%                                             NaN     ...       
75%                                             NaN     ...       
max                                             NaN     ...       

                Appellation1    Appellation2 Appellation3 Appellatio

## based on above, dropping:
- unnamed
- url
- wine type
- varietal

In [4]:
df = df.drop(columns=['Unnamed: 0','Url','wine_type','Varietals'])
df.head()

Alcohol                                   Appellation Category  Points  \
0     14.9  Russian River Valley, Sonoma, California, US      Red      87   
1     14.5  Central Coast, Central Coast, California, US      Red      87   
2     13.8                           California Republic      Red      86   
3     13.8          Lodi, Central Valley, California, US      Red      86   
4     13.0                                New Jersey, US      Red      86   

   Price                                             Review     Variety  \
0  42.00  Baked plum, licorice and lavender aromas and f...   Zinfandel   
1  19.98  Pomegranate and light baking-spice aromas show...  Pinot Noir   
2  15.00  Butter and vanilla notes dominate the jammy fr...  California   
3  10.00  An aroma like toasted almonds and wood smoke m...  Pinot Noir   
4  27.00  Plum skin and pomegranate lead the nose while ...   Red Blend   

                                                Wine               Winery  \
0  V. Sattui 2015 Gilsson Vineyard Old Vine Zinfa...            V. Sattui   
1  Wente 2014 Coastal Selection Pinot Noir (Centr...                Wente   
2  California Republic 2016 Cabernet Sauvignon (C...  California Republic   
3     Collier Creek 2015 Red Wagon Pinot Noir (Lodi)        Collier Creek   
4                  DiLuca 2016 Rosso Black Label Red               DiLuca   

           Appellation1     Appellation2 Appellation3 Appellation4  \
0  Russian River Valley           Sonoma   California           US   
1         Central Coast    Central Coast   California           US   
2   California Republic              NaN          NaN          NaN   
3                  Lodi   Central Valley   California           US   
4            New Jersey               US          NaN          NaN   

       Region    Year               Region2                 Var  
0         NaN  2015.0  Russian River Valley           Zinfandel  
1         NaN  2014.0         Central Coast          Pinot Noir  
2  California  2016.0            California  Cabernet Sauvignon  
3         NaN  2015.0                  Lodi          Pinot Noir  
4         NaN  2016.0                   NaN           Red Blend

In [5]:
df.describe(include= 'all')

Alcohol                                   Appellation Category  \
count   4768.000000                                          4768     4768   
unique          NaN                                           945        7   
top             NaN  Russian River Valley, Sonoma, California, US      Red   
freq            NaN                                           214     3118   
mean      13.949778                                           NaN      NaN   
std        0.950407                                           NaN      NaN   
min        8.000000                                           NaN      NaN   
25%       13.500000                                           NaN      NaN   
50%       14.100000                                           NaN      NaN   
75%       14.500000                                           NaN      NaN   
max       22.000000                                           NaN      NaN   

             Points        Price  \
count   4768.000000  4768.000000   
unique          NaN          NaN   
top             NaN          NaN   
freq            NaN          NaN   
mean      89.340185    42.290271   
std        2.784258    32.311230   
min       80.000000     5.000000   
25%       87.000000    24.000000   
50%       90.000000    35.000000   
75%       91.000000    50.000000   
max      100.000000   475.000000   

                                                   Review     Variety  \
count                                                4768        4768   
unique                                               4678         228   
top     This is an attractive well-made dry wine that ...  Pinot Noir   
freq                                                    2         862   
mean                                                  NaN         NaN   
std                                                   NaN         NaN   
min                                                   NaN         NaN   
25%                                                   NaN         NaN   
50%                                                   NaN         NaN   
75%                                                   NaN         NaN   
max                                                   NaN         NaN   

                                               Wine    Winery  \
count                                          4768      4768   
unique                                         4676      1520   
top     Naggiar 2017 Estate Rosé (Sierra Foothills)  Maryhill   
freq                                              3        38   
mean                                            NaN       NaN   
std                                             NaN       NaN   
min                                             NaN       NaN   
25%                                             NaN       NaN   
50%                                             NaN       NaN   
75%                                             NaN       NaN   
max                                             NaN       NaN   

                Appellation1    Appellation2 Appellation3 Appellation4  \
count                   4768            3482         3477         3384   
unique                   943              32            5            1   
top     Russian River Valley   Central Coast   California           US   
freq                     214             780         2207         3384   
mean                     NaN             NaN          NaN          NaN   
std                      NaN             NaN          NaN          NaN   
min                      NaN             NaN          NaN          NaN   
25%                      NaN             NaN          NaN          NaN   
50%                      NaN             NaN          NaN          NaN   
75%                      NaN             NaN          NaN          NaN   
max                      NaN             NaN          NaN          NaN   

            Region         Year     Region2         Var  
count         1185  4768.000000      

## For Scaling:
- Alcohol
- Points
- Price

## for dummy variables:
- appellation (945 unique)
- category (7 unique)
- var (117 unique)

## NLTK for:
- Reviews
- Wine
- Winery

## Rename:
- Appellation = full address (e.g. Russian River Valley, Sonoma, California, US)
- Appellation 1-4 (this is made up of the appellation split up into columns)
- look at appellation 3 vs region (what's the difference)
- Region vs Region 2


# Appellation_4

In [6]:
df.Appellation4.value_counts(normalize=False, sort=True, ascending=True, bins=None, dropna=False)

NaN    1384
 US    3384
Name: Appellation4, dtype: int64

In [7]:
df.Appellation3.value_counts(normalize=False, sort=True, ascending=True, bins=None, dropna=False)

 US              93
 New York       126
 Washington     507
 Oregon         544
NaN            1291
 California    2207
Name: Appellation3, dtype: int64

In [8]:
df.Appellation2.value_counts(normalize=False, sort=True, ascending=True, bins=None, dropna=False)

 California              1
 Michigan                1
 Pennsylvania            1
 New Jersey              2
 North Carolina          2
 New Mexico              2
 Indiana                 3
 Idaho                   4
 Arizona                 4
 US                      5
 New York Other          5
 Colorado               10
 Texas                  11
 Long Island            16
 Washington Other       17
 Lake County            24
 South Coast            33
 North Coast            43
 Southern Oregon        47
 Virginia               52
 Oregon Other           55
 Central Valley         74
 Napa-Sonoma            89
 California Other       92
 Mendocino County       96
 Finger Lakes          105
 Sierra Foothills      133
 Napa                  271
 Willamette Valley     442
 Columbia Valley       490
 Sonoma                572
 Central Coast         780
NaN                   1286
Name: Appellation2, dtype: int64

## Based on above, will delete Appellation_4 (since all are US) and rename Appellation 3 as STATE

In [9]:
df = df.drop(columns=['Appellation4'])

In [10]:
df = df.rename(columns={'Appellation3': 'State'}) 

In [11]:
df.Appellation2.value_counts(normalize=False, sort=True, ascending=True, bins=None, dropna=False)

 California              1
 Michigan                1
 Pennsylvania            1
 New Jersey              2
 North Carolina          2
 New Mexico              2
 Indiana                 3
 Idaho                   4
 Arizona                 4
 US                      5
 New York Other          5
 Colorado               10
 Texas                  11
 Long Island            16
 Washington Other       17
 Lake County            24
 South Coast            33
 North Coast            43
 Southern Oregon        47
 Virginia               52
 Oregon Other           55
 Central Valley         74
 Napa-Sonoma            89
 California Other       92
 Mendocino County       96
 Finger Lakes          105
 Sierra Foothills      133
 Napa                  271
 Willamette Valley     442
 Columbia Valley       490
 Sonoma                572
 Central Coast         780
NaN                   1286
Name: Appellation2, dtype: int64

In [12]:
df.Appellation2.fillna(df.Appellation1, inplace=True)

In [13]:
df.State.fillna(df.Appellation2, inplace=True)

In [14]:
df.State.value_counts(normalize=False, sort=True, ascending=True, bins=None, dropna=False)

Wind Rose                            1
Refugio Ranch                        1
Erath                                1
La Follette                          1
Divum                                1
Bell                                 1
Stewart                              1
Ex                                   1
Broadley                             1
Stephanie                            1
Knudsen                              1
Coeur de Terre                       1
Adaptation                           1
Seven Daughters                      1
Sineann                              1
Le Vigne                             1
Fulkerson                            1
Starside                             1
Oak Farm                             1
Vellichor                            1
Stevens                              1
Cedar + Salmon                       1
Sweet Heart                          1
Château Souverain                    1
Burnt Bridge                         1
Mercer                   

In [15]:
df = df.rename(columns={'Appellation2': 'Broad_Region'}) 
df = df.rename(columns={'Region': 'Broad_Region'}) 

In [16]:
df = df.rename(columns={'Appellation1': 'Town'}) 

In [17]:
df.describe(include= 'all')

Alcohol                                   Appellation Category  \
count   4768.000000                                          4768     4768   
unique          NaN                                           945        7   
top             NaN  Russian River Valley, Sonoma, California, US      Red   
freq            NaN                                           214     3118   
mean      13.949778                                           NaN      NaN   
std        0.950407                                           NaN      NaN   
min        8.000000                                           NaN      NaN   
25%       13.500000                                           NaN      NaN   
50%       14.100000                                           NaN      NaN   
75%       14.500000                                           NaN      NaN   
max       22.000000                                           NaN      NaN   

             Points        Price  \
count   4768.000000  4768.000000   
unique          NaN          NaN   
top             NaN          NaN   
freq            NaN          NaN   
mean      89.340185    42.290271   
std        2.784258    32.311230   
min       80.000000     5.000000   
25%       87.000000    24.000000   
50%       90.000000    35.000000   
75%       91.000000    50.000000   
max      100.000000   475.000000   

                                                   Review     Variety  \
count                                                4768        4768   
unique                                               4678         228   
top     This is an attractive well-made dry wine that ...  Pinot Noir   
freq                                                    2         862   
mean                                                  NaN         NaN   
std                                                   NaN         NaN   
min                                                   NaN         NaN   
25%                                                   NaN         NaN   
50%                                                   NaN         NaN   
75%                                                   NaN         NaN   
max                                                   NaN         NaN   

                                               Wine    Winery  \
count                                          4768      4768   
unique                                         4676      1520   
top     Naggiar 2017 Estate Rosé (Sierra Foothills)  Maryhill   
freq                                              3        38   
mean                                            NaN       NaN   
std                                             NaN       NaN   
min                                             NaN       NaN   
25%                                             NaN       NaN   
50%                                             NaN       NaN   
75%                                             NaN       NaN   
max                                             NaN       NaN   

                        Town    Broad_Region        State Broad_Region  \
count                   4768            4768         4768         1185   
unique                   943             800          773          122   
top     Russian River Valley   Central Coast   California   California   
freq                     214             780         2207          107   
mean                     NaN             NaN          NaN          NaN   
std                      NaN             NaN          NaN          NaN   
min                      NaN             NaN          NaN          NaN   
25%                      NaN             NaN          NaN          NaN   
50%                      NaN             NaN          NaN          NaN   
75%                      NaN             NaN          NaN          NaN   
max                      NaN             NaN          NaN          NaN   

               Year     Region2         Var  
count   4768.000000        3494        4768  
uni

In [18]:
df.Region2.value_counts(normalize=False, sort=True, ascending=True, bins=None, dropna=False)

Yadkin Valley                   1
Fiddletown                      1
Paicines                        1
Outer Coastal Plain             1
Port                            2
Texas High Plains               2
San Francisco Bay               2
High Valley                     2
Yorkville Highlands             3
Grand Valley                    3
Mt. Harlan                      3
Rockpile                        3
Diamond Mountain District       4
Mount Veeder                    4
America                         4
Dunnigan Hills                  4
Arizona                         4
San Luis Obispo County          5
Idaho                           5
Red Hills                       5
Chalk Hill                      6
San Benito County               6
Coombsville                     7
Suisun Valley                   7
Colorado                        7
Spring Mountain District        8
St. Helena                      8
Texas                           9
Oak Knoll District              9
Clarksburg    

In [21]:
df= df.drop(columns=['Region2'])

In [22]:
df.head()

Alcohol                                   Appellation Category  Points  \
0     14.9  Russian River Valley, Sonoma, California, US      Red      87   
1     14.5  Central Coast, Central Coast, California, US      Red      87   
2     13.8                           California Republic      Red      86   
3     13.8          Lodi, Central Valley, California, US      Red      86   
4     13.0                                New Jersey, US      Red      86   

   Price                                             Review     Variety  \
0  42.00  Baked plum, licorice and lavender aromas and f...   Zinfandel   
1  19.98  Pomegranate and light baking-spice aromas show...  Pinot Noir   
2  15.00  Butter and vanilla notes dominate the jammy fr...  California   
3  10.00  An aroma like toasted almonds and wood smoke m...  Pinot Noir   
4  27.00  Plum skin and pomegranate lead the nose while ...   Red Blend   

                                                Wine               Winery  \
0  V. Sattui 2015 Gilsson Vineyard Old Vine Zinfa...            V. Sattui   
1  Wente 2014 Coastal Selection Pinot Noir (Centr...                Wente   
2  California Republic 2016 Cabernet Sauvignon (C...  California Republic   
3     Collier Creek 2015 Red Wagon Pinot Noir (Lodi)        Collier Creek   
4                  DiLuca 2016 Rosso Black Label Red               DiLuca   

                   Town         Broad_Region                State  \
0  Russian River Valley               Sonoma           California   
1         Central Coast        Central Coast           California   
2   California Republic  California Republic  California Republic   
3                  Lodi       Central Valley           California   
4            New Jersey                   US                   US   

  Broad_Region    Year                 Var  
0          NaN  2015.0           Zinfandel  
1          NaN  2014.0          Pinot Noir  
2   California  2016.0  Cabernet Sauvignon  
3          NaN  2015.0          Pinot Noir  
4          NaN  2016.0           Red Blend

In [24]:
df.to_csv('Cleaned Data with States.csv')